### ISSUE TITLE
Allow extending BarSeriesManager by making fields transactionCostModel and holdingCostModel accessible.

### ISSUE THREAD

**Issue by chrisbu** [ta4j/ta4j#963]
**I have checked existing issues and wiki**
- [X] I could not find similar [issues](https://github.com/ta4j/ta4j/issues?utf8=%E2%9C%93&q=)
- [X] I could not find a solution in the [wiki](https://ta4j.github.io/ta4j-wiki/) or [faq section](https://ta4j.github.io/ta4j-wiki/FAQ.html)

**Is your feature request related to a problem? Please describe.**
I have a requirement to extend the BarSeriesManager, providing a tweaked implementation of `BarSeriesManager#run(...)`.  Unfortunately, as the fields `BarSeriesManager#transactionCostModel` and `BarSeriesManager#holdingCostModel` are private and there are no getters, I'm having to maintain my own fork so that I can extend `BarSeriesManager` and use them when creating a new `TradingRecord`.  

**Describe the solution you'd like**
Given that there is a `getBarSeries()` which returns the private barSeries, I propose that there should also be `getTransactionCostModel()` and `getHoldingCostModel()`, which provides everything needed to allow extending the class and re-implementing `run()`.

**Describe alternatives you've considered**
Alternative would be just to make these two fields `protected` instead of `private`.

**Additional context**
Happy to provide a PR (Essentially this: https://github.com/chrisbu/ta4j/compare/develop...feature/allow_extend_barseriesmanager )  Do I need to add unit test for getters?

**Comment by TheCookieLab** [comment]
Yes, please create a PR. Up to you on the unit tests but if you're simply returning the values then they are not necessary.
